components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/AuthenticationHandler.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.handlers;
+
+import org.apache.axiom.soap.SOAP11Constants;
+import org.apache.axiom.soap.SOAP12Constants;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.description.HandlerDescription;
+import org.apache.axis2.description.Parameter;
+import org.apache.axis2.engine.Handler;
+import org.apache.axis2.namespace.Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.ws.security.WSConstants;
+import org.apache.ws.security.WSSecurityException;
+import org.apache.ws.security.util.Base64;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.handlers.invoker.RESTInvoker;
+import org.wso2.carbon.apimgt.handlers.invoker.RESTResponse;
+import org.wso2.carbon.apimgt.handlers.utils.AuthConstants;
+import org.wso2.carbon.apimgt.handlers.utils.CoreUtils;
+
+import javax.xml.namespace.QName;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AuthenticationHandler implements Handler {
+    private static final Log log = LogFactory.getLog(AuthenticationHandler.class);
+    private static HandlerDescription EMPTY_HANDLER_METADATA = new HandlerDescription("API Security Handler");
+    private HandlerDescription handlerDesc;
+    private ArrayList<String> apiList;
+    private RESTInvoker restInvoker;
+
+    /**
+     * Setting up configurations at the constructor
+     */
+    public AuthenticationHandler() {
+        log.info("Engaging API Security Handler");
```

Comment:
As discussed offline, the log only prints at server startup and doesn't trigger upon every request.
It is added to notify developers through the log inspection for any future support analysis. Therefore added a comment in commit: 33422cb